File: src/test/python/sample/ManageUsersTest.py
import unittest
from sample.ManageUsers import create_new_user
from src.main.proto.sample import user_profile_pb2

class TestManageUsers(unittest.TestCase):

    def test_create_user_profile(self):
        user_id: str = "py_test_001"
        name: str = "Python Test User Proto"
        age: int = 28
        is_active: bool = True
        interests: list[str] = ["testing", "python", "protobuf"]

        profile = create_new_user(user_id, name, age, is_active, interests)

        self.assertIsInstance(profile, user_profile_pb2.UserProfile)
        self.assertEqual(profile.user_id, user_id)
        self.assertEqual(profile.display_name, name)
        self.assertEqual(profile.age, age)
        self.assertEqual(profile.is_active, is_active)
        self.assertListEqual(list(profile.interests), interests)

if __name__ == '__main__':
    unittest.main()

File: src/test/kotlin/sample/ManageUsersTest.kt
package sample

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import sample.proto.UserProfile 
import sample.createKotlinUser

@RunWith(JUnit4::class)
class ManageUsersTest {

    @Test
    fun `createKotlinUser should populate fields correctly`() {
        val userId = "kt_test_001"
        val displayName = "Kotlin Proto Test User"
        val age = 30
        val isActive = true
        val interests = listOf("kotlin", "protobuf", "bazel", "testing")

        val profile: UserProfile = createKotlinUser(userId, displayName, age, isActive, interests)
        assertEquals(userId, profile.userId)
        assertEquals(displayName, profile.displayName)
        assertEquals(age, profile.age)
        assertEquals( isActive, profile.isActive)
        assertEquals(interests.size, profile.interestsList.size)
        assertTrue(profile.interestsList.containsAll(interests) && interests.containsAll(profile.interestsList))
    }

}